src/ansys/sound/core/signal_processing/filter.py

```diff
@@ -0,0 +1,311 @@
+
+        self.__sampling_frequency = sampling_frequency
+
+        if file != "":
+            if a_coefficients != [] or b_coefficients != []:
+                warnings.warn(
+                    PyAnsysSoundWarning(
+                        "Specified parameters a_coefficients and b_coefficients are ignored "
+                        "because FRF file is also specified."
+                    )
+                )
+            self.design_FIR_from_FRF_file(file)
+        else:
+            self.a_coefficients = a_coefficients
+            self.b_coefficients = b_coefficients
+
+        self.signal = signal
+
+    def __str__(self) -> str:
+        """Return the string representation of the object."""
+        if len(self.a_coefficients) > 5:
+            str_a = str(self.a_coefficients[:5])[:-1] + ", ... ]"
+        elif len(self.a_coefficients) == 0:
+            str_a = "Not set"
+        else:
+            str_a = str(self.a_coefficients)
+
+        if len(self.b_coefficients) > 5:
+            str_b = str(self.b_coefficients[:5])[:-1] + ", ... ]"
+        elif len(self.b_coefficients) == 0:
+            str_b = "Not set"
+        else:
+            str_b = str(self.b_coefficients)
+
+        return (
+            f"Sampling frequency: {self.__sampling_frequency:.1f} Hz\n"
+            f"Numerator coefficients (B): {str_b}\n"
+            f"Denominator coefficients (A): {str_a}"
+        )
+
+    @property
+    def a_coefficients(self) -> list[float]:
+        """Denominator coefficients of the filter's transfer function."""
+        return self.__a_coefficients
+
+    @a_coefficients.setter
+    def a_coefficients(self, coefficients: list[float]):
+        """Set filter's denominator coefficients."""
+        self.__a_coefficients = coefficients
+
+    @property
+    def b_coefficients(self) -> list[float]:
+        """Numerator coefficients of the filter's transfer function."""
+        return self.__b_coefficients
+
+    @b_coefficients.setter
+    def b_coefficients(self, coefficients: list[float]):
+        """Set filter's numerator coefficients."""
+        self.__b_coefficients = coefficients
+
+    @property
+    def signal(self) -> Field:
+        """Input signal as a DPF field."""
+        return self.__signal
+
+    @signal.setter
+    def signal(self, signal: Field):
+        """Set input signal."""
+        if signal is not None:
+            if not (isinstance(signal, Field)):
+                raise PyAnsysSoundException("Specified signal must be provided as a DPF field.")
+
+            time_data = signal.time_freq_support.time_frequencies.data
+            if len(signal.data) < 2 or len(time_data) < 2:
+                raise PyAnsysSoundException("Specified signal must have at least two samples.")
+
+            signal_fs = 1 / (time_data[1] - time_data[0])
+            if np.round(signal_fs, 1) != np.round(self.__sampling_frequency, 1):
+                raise PyAnsysSoundException(
```

Comment:
@ansaminard we're not as strict in the DPF Sound operator, I'd like to have the input of @anshlachamb and @anssmolla regarding that `if` statement

Because technically, we can still filter if the sampling frequency of the signal is different, right ? 

Maybe just a warning would be enough ?

Reply:
@a-bouth But then the result would be wrong...
The sampling frequency is linked to the filter coefficients. If we filter a signal with a different fs, then the coefficients are no longer adapted, and the produced frequency response will not be the expected one
A warning would be adapted if we would resample the signal as a consequence

Reply:
@ansaminard I understand, but I'm more in favor of letting the user do mistakes at his own expanses rather than preventing him of doing something. 

Maybe for whatever weird scenario, the user would like to use coefficients with a different sampling frequency than the signal 

The point of PyAnsys Sound is to give more flexibility to the user, in my opinion, it's not supposed to be as strict and as "we take the user by that hand" as SAS

Reply:
@a-bouth I feel this in particular is too big of a mistake to let the user do it. I could really lead to hidden errors, if they do not pay attention to warnings (which, admit it, is quite often something _we_ do ourselves). Imagine a situation where you have a large workflow and get a lot of warnings that you can generally ignore (for whatever reason), this warning might go unnoticed.
If the user reeeally wants to filter a signal with a different fs and know what they're doing, they can still do it, just not with this class.

Anyway, that's my opinion, if @anssmolla and @anshlachamb also think it should be a warning, I'll do the change.

Reply:
@a-bouth @ansaminard 
I would agree with @ansaminard: this is a too big hidden mistake for us to let the user do it.
If for whatever reason the user wants to filter a signal with a different Fs, then he will be able to create a new filter with same coeffs and the correct Fs to do the filtering

@anssmolla ?

Reply:
discussed with @anssmolla : "as we have a Fs in the Filter object, then we do a strict check". So we do not let the user filter a signal with a different Fs. Let's keep this code part as is
@ansaminard